### PR TITLE
fix: datadog injection error

### PR DESCRIPTION
**Pull request by lodmfjord**
# Fix DataDog injection error.

dd-trace patches a few things like: require and promises, adding info for tracing.

This breaks a few tests - for example, when comparing promise objects.

## What

Making test pass after adding dd-trace.

## Why

dd-trace changes how jest is run, which makes tests fail when input is an array and when comparing promise objects. 

## How

- Check if promises are called instead of comparing them.
- Patch dd-trace to not use .shift
- Fix race condition on stopping postgres
- Add dd-trace running locally.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by eirikurn — COMMENTED** [review]
These fixes look really hacky and these errors will likely stall developers in the future.

The tests just looked like basic jest tests. Does datadog have documentation on how to write jest tests to avoid issues like these?

**Comment by lodmfjord** [comment]
@eirikurn 
> 



> These fixes look quite hacky. I imagibe developers will run into these errors again in the future and be stuck.
> 
> Does datadog have documentation on how to write jest tests to avoid issues like these?

I agree.
I just needed to make sure why the errors occurred. 

Not hackish:
a) apps/api/src/app/graphql.middleware.spec.ts - this fix is better since comparing promise objects should not be done.
b) libs/testing/containers/src/lib/testing-containers.ts - not returning a promise here created a race condition, which sometimes created problems. 

Very hackish:
a) apps/judicial-system/backend/src/app/modules/case/filters/case.filter.spec.ts 
b) apps/services/auth/delegation-api/src/app/delegations/test/me-delegations.access-incoming.spec.ts 

Are hackish to confirm the error. Going to work on it.

I have read the documentation and all issues.

**Comment by lodmfjord** [comment]
@eirikurn 
const testParameters = parametersByTestName[testName].shift();

This caused arrays in input to test to be cleared out - which was only the case for judicial-system and services-auth-delegation-api.

Weird going to see how my patch works and if it changes output too much.

**Comment by lodmfjord** [comment]
@eirikurn 
I also added that dd-trace is run locally (local env, so no info gets pushed), so errors like this should be found quicker.

I will create an issue on dd-trace about this "destructive" behaviour to see if we are doing something anti-pattern.

**Review by eirikurn — COMMENTED** [review]
Nice job. Just one question.

File: dd-trace-run/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "dd-trace-run",
+  "version": "0.0.0.1",
+  "packageManager": "yarn@3.2.3",
+  "main": "./run.js"
```
[the remark this answers is not in the document]
Comment (eirikurn) [reply dd-trace-run/package.json]: Already resolved

File: .yarn/patches/node-request-interceptor-npm-0.5.9-77e9d9c058.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+diff --git a/lib/interceptors/ClientRequest/ClientRequestOverride.js b/lib/interceptors/ClientRequest/ClientRequestOverride.js
+index e1e005c1055587ed12d1084dab3821b93572ca3d..9de4403259845b19c802b0677a8db9f71442bc26 100644
+--- a/lib/interceptors/ClientRequest/ClientRequestOverride.js
++++ b/lib/interceptors/ClientRequest/ClientRequestOverride.js
+@@ -1,3 +1,5 @@
++// FIX: jsdom does not have nodejs utils
++global.setImmediate = global.setImmediate || ((fn, ...args) => global.setTimeout(fn, 0, ...args));
```
[the remark this answers is not in the document]
Comment (lodmfjord) [reply .yarn/patches/node-request-interceptor-npm-0.5.9-77e9d9c058.patch:7]: This is something wrong in dd-trace should fix that - I will add it there. Good point.

File: dd-trace-run/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (lodmfjord) [reply dd-trace-run/package.json]: I should remove this. No longer needed. Accident.

File: .yarn/patches/node-request-interceptor-npm-0.5.9-77e9d9c058.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (lodmfjord) [reply .yarn/patches/node-request-interceptor-npm-0.5.9-77e9d9c058.patch]: ok. done.

[the remark this answers is not in the document]
Comment (lodmfjord) [reply .yarn/patches/node-request-interceptor-npm-0.5.9-77e9d9c058.patch]: updating branch.

**Review by eirikurn — APPROVED** [review]
Awesome! Can you post the upstream dd-trace issue here for reference?

**Comment by lodmfjord** [comment]
Related dd-trace issue:
https://github.com/DataDog/dd-trace-js/issues/3504

**Pull request merged**